### ISSUE TITLE
Handle a layout for both node views and node revisions view

### ISSUE THREAD

**Issue by herbdool** [backdrop-contrib/layout_wildcard#32]
I was thinking that this module would work to handle a layout for both node views and node revisions view. So both `node/%` and `node/%/revisions/%/view`. The validation prevents it from saving since the number of placeholders are supposed to match.

I figure that in at least this case so long as the number of placeholders in the alternative paths is the same or more than it should be okay. It does work when I allow the alternative paths to have more placeholders.

Perhaps it's fine as long as it's clear that only the first placeholder is used to load context? Is there a downside? The other placeholders are, by default, passed through as strings.

I also tried to use "ancestor matching" to create this (if I understand the feature correctly). Made the `node/%/revisions/%/view` the primary path and selected ancestor matching.

**Comment by bugfolder** [comment]
Yeah, I've been thinking about adding a checkbox, "strict context matching", to allow this scenario.

**Comment by herbdool** [comment]
In my scenario it would need to strictly match the first placeholder but not the rest. Does your proposed option do more than just count the number of placeholders?

**Comment by bugfolder** [comment]
What I was thinking was with "strict matching" on, the number and order of placeholders would have to match, and with "strict matching" off, only placeholders that were defined with non-string context values would have to be matched and the rest passed through as strings. Would that work for you?

Alternative logic proposals would be OK. We just have to make sure any additions or changes don't break existing usages.

**Comment by herbdool** [comment]
I don't have an alternative at the moment. I wonder what others think.

I don't know if I'll be using this module immediately, but I got down this path from the core issue: https://github.com/backdrop/backdrop-issues/issues/1528 and I was wondering how layout_wildcard implemented it. Then I was surprised to see that it used placeholders too and not "wildcards" to match on anything. So it works pretty similarly to what @quicksketch suggested a few years ago on that issue. So maybe some coordination to get some of this module's functionality into core, is needed.

**Comment by bugfolder** [comment]
@herbdool, you could also check out the Roadmap I laid out for this module:

https://github.com/backdrop-contrib/layout_wildcard/issues/19

I planned to handle true * wildcards, also aliases, but got sidetracked before implementing them. If this is going to get attention in core, perhaps I'll hold off on it. https://github.com/backdrop/backdrop-issues/issues/1602 also had some discussion of the latter.

**Comment by indigoxela** [comment]
> https://github.com/backdrop/backdrop-issues/issues/1602 also had some discussion of the latter

Not really. You might have mixed it up with https://github.com/backdrop/backdrop-issues/issues/1528, which - to me - reads like a "no special wildcards but multiple paths in core". :wink: Sort of a "light variant" without the tricky traps the other issue ran into https://github.com/backdrop/backdrop-issues/issues/1602

**Comment by izmeez** [comment]
I haven't considered the whole `wildcard` issue. However, the module works great for adding additional paths rather than cloning the display for common content paths: `node/%`, `node/%/edit`, `node/%/revisions` and `node/%/revisions/view`.
